lua/includes/modules/callable_properties.lua

```diff
@@ -0,0 +1,73 @@
+CallableProperties = {}
+
+local function setup(object)
+    local meta = getmetatable(object)
+
+    -- Handles situations where the metatabale's __index was already set
+    local originalIndex
+    if meta then
+        originalIndex = meta.__index
+    end
+
+    setmetatable(object, {
+        __callableProperties = {},
+        __originalIndex = originalIndex,
+        __index = function (self, index)
```

Comment:
The `__index` method is called when a table is accessed with a non-existent key. We abuse that by removing the key from the actual table and storing its value here in the metatable, so when people go to access their registered key it gets routed through here instead.